packages/headless/src/controllers/knowledge/generatedAnswer/headless-knowledge-generated-answer.ts

```diff
@@ -42,6 +42,25 @@ export type EngineNeededForKnowledgeGeneratedAnswer = CoreEngine<
     DebugSection & {answer: ReturnType<typeof answerApi.reducer>}
 >;
 
+const subscribeToSearchRequest = (
+  engine: EngineNeededForKnowledgeGeneratedAnswer
+) => {
+  let lastRequestId: string;
```

Comment:
This will retrigger a new `fetchAnswer()` whenever `requestID` changes: 

Should it instead be "whenever `requestID` + `q` changes?"

For example, if the user "loads more results" (pager), but don't change the intent of the query (`q` does not change), I don't think it's meaningful to resend a request to `fetchAnswer()` ?

Here I talk about pager, but it could be facetting change, sorting changes etc.

Reply:
This can be done later by the way;

I have a hunch the best approach will be to have a shortlist of parameters that, even if they do end up requesting a new request to search API when they change, that they should not automatically 1 to 1 retrigger a gen ai request.

Reply:
^ I would advise creating a (memoized) selector that returns those parameters in an object (or a hash, tho I think the former would be more efficient) and do a reference (=== or Object.is) comparison on the previous value returned from the selector and the new one.

You'll be able just to add parameters as required as you go easily.

Reply:
I will apply the suggestion and create a selector with the triggerParams because you guys are right about it.

It is good to know though that a requestId change with the same `q` wouldn't have called the endpoint again because of the infamous caching mechanism of RTK query.